api/products.js

```diff
@@ -0,0 +1,45 @@
+const router = require('express').Router();
+const {Order, Product, LineItem, User, Category} = require('../db').models;
+
+router.get('/', (req, res, next) => {
+    Product.findAll({
+        include: [Category] 
+    })
+    .then(products => {
+        res.send(products)
+    })
+    .catch(next)
+})
+
+router.get('/:productId', (req, res, next) => {
+    Product.findById(req.params.productId*1, {
+        include: [Category]
+    })
+    .then(product => res.send(product))
+    .catch(next)
+})
+
+router.post('/', (req, res, next) => {
+    Product.create(req.body)
+    .then(product => res.send(product))
+    .catch(next)
+})
+
+router.put('/:productId', (req, res, next) => {
+    Product.findById(req.params.productId*1)
+    .then(product => {
+        product.update(req.body)
```

Comment:
In api/lineItems.js, Tim and I had a different pattern for updating.
Instead of doing a product.update with the whole req.body, we set new values to applicable fields and saved.
I have a hunch it would be better to use the same pattern, as it may influence how we send update information from the front end to the back end?